lib/tpm2_eventlog.c

```diff
@@ -421,16 +421,18 @@ bool specid_event(TCG_EVENT const *event, size_t size,
 
 bool parse_eventlog(tpm2_eventlog_context *ctx, BYTE const *eventlog, size_t size) {
 
-    TCG_EVENT_HEADER2 *next;
-    TCG_EVENT *event = (TCG_EVENT*)eventlog;
-    bool ret;
+    if(!eventlog || (sizeof eventlog < sizeof(TCG_EVENT))) {
+        return false;
+    }
 
+    TCG_EVENT *event = (TCG_EVENT*)eventlog;
     if (!event) {
         return false;
     }
 
     if (event->eventType == EV_NO_ACTION) {
```

Comment:
why is this a valid access?— in that it is neither the effective type nor character type.

Reply:
> why is this a valid access?— in that it is neither the effective type nor character type.

TCG_event is a packed structure.